### DISCUSSION TITLE
Warning/info if child class re-implements interface already implemented by base class

### DISCUSSION THREAD

**Discussion by TahirAhmadov** [dotnet/csharplang#6244] [Language Ideas]
Does it make sense to create an info message (a warning is probably a breaking change and too harsh) when a child class implements an interface which is already implemented by a base class? While not a real problem by itself, it's likely an indication of a mistake being made; there is a strong code smell from this. However, I'm not entirely sure - perhaps there is a valid reason to do so - if so, any input would be appreciated.

**Comment by HaloFour** [comment]
This has always been a legal thing to do, and it's a perfectly legit way to enable a derived class provide an alternate implementation of the given interface.

**Reply by TahirAhmadov** [reply]
Good point. I almost think that for this use case, `new` keyword should be used, to indicate that this is a new implementation. Also, it should then require a new set of members to be defined - piggying back to base class members shouldn't be allowed (implicitly - you can call them, as needed, from child class's version of those members).

**Reply by HaloFour** [reply]
I'm thinking we're about 20 years too late for that discussion.  This feature has always existed and it has never been considered an anti-pattern.
